libamiibo/Data/AmiiboTag.cs

```diff
@@ -199,6 +203,94 @@ public static AmiiboTag FromInternalTag(ArraySegment<byte> data)
             return new AmiiboTag(data) { IsDecrypted = true };
         }
 
+        /// <summary>
+        /// Creates an AmiiboTag instance from an identification block represented as a hexadecimal string.
+        /// </summary>
+        /// <param name="identificationBlock">The identification block as a hexadecimal string.</param>
+        /// <returns>An AmiiboTag instance representing the identification block.</returns>
+        public static AmiiboTag FromIdentificationBlock(string identificationBlock)
+        {
+            // Create a regular expression object to match a 16-character hexadecimal string..
+            Regex regex = new Regex(@"^(?:0x)?([0-9A-Fa-f]{16})$");
+
+            // Match the input identification block against the regular expression.
+            Match match = regex.Match(identificationBlock);
+
+            if (match.Success)
+            {
+                // The captured group at index 1 contains the hexadecimal value.
+                string hexadecimalValue = match.Groups[1].Value;
+
+                // Call the method to create an AmiiboTag from a byte array.
+                return FromIdentificationBlock(ByteHelpers.StringToByteArray(hexadecimalValue));
+            }
+            else
+            {
+                // The input identification block does not match the expected format.
+                throw new ArgumentException(nameof(identificationBlock));
+            }
+        }
+
+        /// <summary>
+        /// Creates an AmiiboTag instance from an identification block represented as a byte array.
+        /// </summary>
+        /// <param name="identificationBlock">The identification block as a byte array.</param>
+        /// <returns>An AmiiboTag instance representing the identification block.</returns>
+        public static AmiiboTag FromIdentificationBlock(byte[] identificationBlock)
+        {
+            // Check if the identification block byte array is null.
+            if (identificationBlock == null)
+            {
+                throw new ArgumentNullException(nameof(identificationBlock));
+            }
+
+            // Check that the identification block byte array has a length of 8 bytes.
+            if (identificationBlock.Length != 8)
+            {
+                throw new ArgumentOutOfRangeException(nameof(identificationBlock));
+            }
+
+            // Check that the format version of the identification block is equal to 0x02.
+            if (identificationBlock[7] != 0x02)
+            {
+                throw new ArgumentException(nameof(identificationBlock));
+            }
+
+            // Create a new byte array to hold decrypted tag data.
+            var decryptedNtag = new byte[540];
+
+            // Copy the tag header to the byte array.
+            tagHeader.CopyTo(decryptedNtag, 0);
+
+            // Copy the identification block to the tag byte array.
+            identificationBlock.CopyTo(decryptedNtag, 84);
```

Comment:
The rest of the code uses hex for positions. Please adjust to keep it consistent.